Re-add vertical input to intent capture

alshakero opened the pull request:
#### Changes proposed in this Pull Request

This PR re-adds the vertical input to the intent capture.
#### Caveat
When the user presses "I don't know" to the vertical, I couldn't figure out how the UI should behave, and it's not spec'd in the design.

#### Testing

Please test as many as possible scenarios of site vertical and site title variations. If you could test on your phone it would be nice too.

a8ci18n commented on client/landing/gutenboarding/onboarding-block/acquire-intent/vertical-select/index.tsx:
```diff
@@ -0,0 +1,239 @@
+				'vertical-select--with-suggestions': !! suggestions.length && isMobile,
+			} ) }
+		>
+			<label htmlFor="vertical-input">{ __( 'My site is about' ) } </label>
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'My site is about <Input />.' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/vertical-input)

simison commented:
Let's move the underline little lower, so that there's space between the text and underline:

<img width="1268" alt="Screenshot 2020-08-07 at 14 14 43" src="https://user-images.githubusercontent.com/87168/89640588-8da1e180-d8b8-11ea-92f1-62d9bbc1434d.png">


should be more like:

<img width="1053" alt="Screenshot 2020-08-07 at 14 15 39" src="https://user-images.githubusercontent.com/87168/89640582-8b3f8780-d8b8-11ea-8e0f-74ded6fa954d.png">

simison commented:
@olaolusoga hi! could you test this (https://calypso.live/new?branch=add/vertical-input) and give feedback if this looks good?

simison commented:
On mobile there's a stray arrow:

![image](https://user-images.githubusercontent.com/87168/89642174-ede65280-d8bb-11ea-9e01-231c44dc6a54.png)

Spacing with underline feels a bit off for mobile, too: 

![image](https://user-images.githubusercontent.com/87168/89642199-02c2e600-d8bc-11ea-9ed0-daab8a1be06a.png)

At about 750px width, odd things happen:

![image](https://user-images.githubusercontent.com/87168/89642232-179f7980-d8bc-11ea-883d-181540e5b295.png)

If user skips the first step, we could just hide the vertical input instead of keeping both visible? Kinda like how it's on mobile.
![image](https://user-images.githubusercontent.com/87168/89642301-3c93ec80-d8bc-11ea-997b-81e462a12624.png)

olaolusoga commented:
Thanks for the ping @simison 

We should match the v2 build for intent capture. Video can be found on the release & result post, P2-pbAok1-16b.

Some notes from comparing the two:
- The input line bottom border is very long, and should be gray when there's no input. (see video)
- When a person types the active text should be underlined with the blue, this controls the length of the bottom border.
- The second field shouldn't have a bottom border, but should keep the placeholder text animation.
- "I don't know" should be in the white bg/gray border button (https://d.pr/i/2gON5M), and when it changes to "Choose a design" (blue) it should be the same width and height (match styling in video)
- The suggestion pop out shouldn't overlay the border line, it should be below it (https://d.pr/i/F6qYpq, see video for reference)

razvanpapadopol commented:
Few notes after a quick run:

### Accessibility
* Tabbing or pressing _Enter_ doesn't work when both inputs are displayed and you're focused on the first one.
* On mobile there is no way to go back to vertical screen once you advanced to site title screen. I remember there was some arrow.
<img width="562" alt="Screenshot 2020-08-07 at 18 06 47" src="https://user-images.githubusercontent.com/14192054/89660097-88a15a00-d8d9-11ea-8804-08fec23a31e4.png">

* Even if the border is full width, the area is not entirely clickable.
<img width="542" alt="Screenshot 2020-08-07 at 18 06 25" src="https://user-images.githubusercontent.com/14192054/89660130-92c35880-d8d9-11ea-9256-12dd476b48e0.png">

* On mobile if you've typed more than 3 characters and Skip button disappears, there is no way to advance to site title screen. There should be a continue button.
<img width="570" alt="Screenshot 2020-08-07 at 18 17 23" src="https://user-images.githubusercontent.com/14192054/89660632-462c4d00-d8da-11ea-8ba1-92698be17936.png">


### Design
* Skip and continue buttons are not styled consistent with the rest of Gutenboarding. Check `@automattic/onboarding` components.
* Overlapping domain suggestions on mobile
<img width="326" alt="Screenshot 2020-08-07 at 18 03 59" src="https://user-images.githubusercontent.com/14192054/89660263-c9996e80-d8d9-11ea-9315-df95de68e621.png">

alshakero commented:
I addressed all feedback, and rebased this and the feature branch. The only discrepancy with the design is the dot after the vertical input. And the "Next" arrow on mobile, I made it show a little lower than it should, it makes the code much much simpler, if this is a deal breaker, please LMK.

simison commented:
Tiny nitpick, the space between line and the nudge needs to be a bit bigger:

### this PR
<img width="792" alt="Screenshot 2020-08-10 at 18 06 03" src="https://user-images.githubusercontent.com/87168/89797981-46278980-db34-11ea-910b-86be0ae335c7.png">

### production

<img width="888" alt="Screenshot 2020-08-10 at 18 06 09" src="https://user-images.githubusercontent.com/87168/89797967-432c9900-db34-11ea-9517-193ec2a4af34.png">

razvanpapadopol commented:
Changing an existing site vertical using the suggestions doesn't work. Tried clicking directly a suggestion and using Enter.
demo: https://cloudup.com/csoMUcDHUmp

alshakero commented:
Addressed all!

simison commented:
There's something funky with longer texts when it doesn't fit anymore (note scroller)

<img width="383" alt="Screenshot 2020-08-10 at 19 14 13" src="https://user-images.githubusercontent.com/87168/89805108-d5856a80-db3d-11ea-9024-fd962e80102b.png">

I'm not sure what I'd expect in that situation — probably the input get capped in `max-width: 100vw;` or so.

I also observed it produce longer underline somehow in some resolution with longer text:

<img width="1077" alt="image" src="https://user-images.githubusercontent.com/87168/89805253-0bc2ea00-db3e-11ea-8860-fffd6ec3e1a1.png">

I don't think these are blockers though, and we can follow-up fix tomorrow.

simison commented:
Note failed `typecheck-strict`

simison commented:
Note on pad-width the scrollbar and placeholder going over:

<img width="1048" alt="Screenshot 2020-08-10 at 19 19 18" src="https://user-images.githubusercontent.com/87168/89805524-707e4480-db3e-11ea-8660-eaff8c5493d5.png">
 
Scrollbar would be great to get rid of, and in current production version I think it caps at input width so this isn't that biggie.

simison commented:
Some odd spacing in this resolution:

<img width="800" alt="image" src="https://user-images.githubusercontent.com/87168/89806522-d7e8c400-db3f-11ea-9aa5-919b166eace8.png">

razvanpapadopol commented:
Something mentioned in https://github.com/Automattic/wp-calypso/pull/44701#issuecomment-670570334 is only partly fixed.
> Tabbing or pressing Enter doesn't work when both inputs are displayed and you're focused on the first one.

Tabbing still doesn't work. 

Also, something that used to work before latest fixes: if you are navigating using arrows through results, you can press _Enter_ to select one of them but you can't proceed anymore with current input by using TAB since this key is now ignored.

razvanpapadopol commented:
Horizontal scroll issues and zoom on mobile were all fixed with https://github.com/Automattic/wp-calypso/pull/44701/commits/c3e8d3f02c727e7e0349b9a2110b2a633b91df29

One small thing I can see on mobile is the suggestions dropdown overlapping _Next_ arrow.
<img width="289" alt="Screenshot 2020-08-10 at 20 03 07" src="https://user-images.githubusercontent.com/14192054/89810115-38c6cb00-db45-11ea-8d4b-a060eabc8b31.png">
